nix/pkg-info.nix

```diff
@@ -0,0 +1,60 @@
+# collects information about a single nixpkgs package
+{ lib
+, pkg
+, ...
+}:
+with builtins;
+assert lib.isDerivation pkg; let
+  # trace with reason
+  trc = info: pkg: trace (info + ": " + toString pkg);
+
+  # if thing is a list, map the function, else apply f to thing and return a singleton of
+  # it
+  mapOrSingleton = f: x:
+    if isList x
+    then map f x
+    else [ (f x) ];
+
+  # things to save from the src attr (the derivation that was created by a fetcher)
+  srcInfo = {
+    urls = (pkg.src.urls or (trc "package didn't have src or url" pkg [ ])) ++ [ (pkg.src.url or null) ];
```

Comment:
`trc "package didn't have src or url" pkg [ ]` isn't the empty list one parameter too much? :thinking:

Reply:
(As far as I understand `trc` takes two arguments.)

Reply:
yes, it takes two arguments but it returns a function! ;)